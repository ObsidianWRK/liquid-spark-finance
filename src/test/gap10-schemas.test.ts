import { describe, it, expect } from "vitest";
import {
  LinkedAccountSchema,
  RecurringChargeSchema,
  NegotiationCaseSchema,
  AutosavePlanSchema,
  HouseholdSchema,
  AgeMetricSchema,
  PrivacySettingSchema,
  AdvisorThreadSchema,
  SpendableCashSchema,
  HomeWidgetSchema,
} from "../schemas/gap10";

/**
 * Simple helper to make a valid ISO date string.
 */
const now = () => new Date().toISOString();

describe("Gap-10 Domain Schemas", () => {
  it("validates LinkedAccount", () => {
    const obj = {
      id: "acc1",
      provider: "mock",
      displayName: "Mock Checking",
      institutionName: "Mock Bank",
      lastFour: "1234",
      type: "checking",
      createdAt: now(),
      updatedAt: now(),
    };
    expect(LinkedAccountSchema.parse(obj)).toMatchSnapshot();
  });

  it("validates RecurringCharge", () => {
    const obj = {
      id: "rc1",
      accountId: "acc1",
      merchantName: "Netflix",
      amount: 15.99,
      frequency: "monthly",
      nextDueDate: now(),
      status: "active",
    };
    expect(RecurringChargeSchema.parse(obj)).toMatchSnapshot();
  });

  it("validates NegotiationCase", () => {
    const obj = {
      id: "case1",
      chargeId: "rc1",
      status: "queued",
      submittedAt: now(),
    };
    expect(NegotiationCaseSchema.parse(obj)).toMatchSnapshot();
  });

  it("validates AutosavePlan", () => {
    const obj = {
      id: "plan1",
      accountId: "acc1",
      targetAmount: 50,
      cadence: "weekly",
      nextTransferDate: now(),
      isActive: true,
    };
    expect(AutosavePlanSchema.parse(obj)).toMatchSnapshot();
  });

  it("validates Household", () => {
    const obj = {
      id: "house1",
      name: "Smith Family",
      members: [
        { userId: "user1", role: "owner" },
        { userId: "user2", role: "member" },
      ],
      createdAt: now(),
    };
    expect(HouseholdSchema.parse(obj)).toMatchSnapshot();
  });

  it("validates AgeMetric", () => {
    const obj = {
      averageDaysHeld: 27,
      calculatedAt: now(),
    };
    expect(AgeMetricSchema.parse(obj)).toMatchSnapshot();
  });

  it("validates PrivacySetting", () => {
    const obj = {
      hideAmounts: true,
      updatedAt: now(),
    };
    expect(PrivacySettingSchema.parse(obj)).toMatchSnapshot();
  });

  it("validates AdvisorThread", () => {
    const obj = {
      id: "thread1",
      userId: "user1",
      messages: [
        { id: "msg1", sender: "user", content: "Hi!", createdAt: now() },
      ],
      isEscalated: false,
    };
    expect(AdvisorThreadSchema.parse(obj)).toMatchSnapshot();
  });

  it("validates SpendableCash", () => {
    const obj = {
      amount: 1200,
      payday: now(),
      calculatedAt: now(),
    };
    expect(SpendableCashSchema.parse(obj)).toMatchSnapshot();
  });

  it("validates HomeWidget", () => {
    const obj = {
      id: "widget1",
      type: "balance",
      position: 1,
      config: {},
    };
    expect(HomeWidgetSchema.parse(obj)).toMatchSnapshot();
  });
}); 