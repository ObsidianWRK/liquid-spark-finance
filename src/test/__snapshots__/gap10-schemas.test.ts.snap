// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Gap-10 Domain Schemas > validates AdvisorThread 1`] = `
{
  "id": "thread1",
  "isEscalated": false,
  "messages": [
    {
      "content": "Hi!",
      "createdAt": "2025-06-19T14:20:41.036Z",
      "id": "msg1",
      "sender": "user",
    },
  ],
  "userId": "user1",
}
`;

exports[`Gap-10 Domain Schemas > validates AgeMetric 1`] = `
{
  "averageDaysHeld": 27,
  "calculatedAt": "2025-06-19T14:20:41.034Z",
}
`;

exports[`Gap-10 Domain Schemas > validates AutosavePlan 1`] = `
{
  "accountId": "acc1",
  "cadence": "weekly",
  "id": "plan1",
  "isActive": true,
  "nextTransferDate": "2025-06-19T14:20:41.032Z",
  "targetAmount": 50,
}
`;

exports[`Gap-10 Domain Schemas > validates HomeWidget 1`] = `
{
  "config": {},
  "id": "widget1",
  "position": 1,
  "type": "balance",
}
`;

exports[`Gap-10 Domain Schemas > validates Household 1`] = `
{
  "createdAt": "2025-06-19T14:20:41.033Z",
  "id": "house1",
  "members": [
    {
      "role": "owner",
      "userId": "user1",
    },
    {
      "role": "member",
      "userId": "user2",
    },
  ],
  "name": "Smith Family",
}
`;

exports[`Gap-10 Domain Schemas > validates LinkedAccount 1`] = `
{
  "createdAt": "2025-06-19T14:20:41.024Z",
  "displayName": "Mock Checking",
  "id": "acc1",
  "institutionName": "Mock Bank",
  "lastFour": "1234",
  "provider": "mock",
  "type": "checking",
  "updatedAt": "2025-06-19T14:20:41.024Z",
}
`;

exports[`Gap-10 Domain Schemas > validates NegotiationCase 1`] = `
{
  "chargeId": "rc1",
  "id": "case1",
  "status": "queued",
  "submittedAt": "2025-06-19T14:20:41.032Z",
}
`;

exports[`Gap-10 Domain Schemas > validates PrivacySetting 1`] = `
{
  "hideAmounts": true,
  "updatedAt": "2025-06-19T14:20:41.035Z",
}
`;

exports[`Gap-10 Domain Schemas > validates RecurringCharge 1`] = `
{
  "accountId": "acc1",
  "amount": 15.99,
  "frequency": "monthly",
  "id": "rc1",
  "merchantName": "Netflix",
  "nextDueDate": "2025-06-19T14:20:41.030Z",
  "status": "active",
}
`;

exports[`Gap-10 Domain Schemas > validates SpendableCash 1`] = `
{
  "amount": 1200,
  "calculatedAt": "2025-06-19T14:20:41.036Z",
  "payday": "2025-06-19T14:20:41.036Z",
}
`;
